frontend/js/pages/DetailBattle.js

```diff
@@ -6,88 +6,99 @@ import styled from 'styled-components';
 import party from '../../image/party-popper.png';
 import { setBattle } from '../actions/setBattle';
 import PokemonCard from '../components/PokemonCard';
+import TrainersInBattle from '../components/TrainersInBattle';
 
-const StyledTitle = styled.span`
+const Title = styled.span`
   font-weight: 600;
 `;
 
-const StyledContainer = styled.div`
+const Container = styled.div`
   margin-top: 30px;
 `;
 
-const StyledRoundContainer = styled.div`
+const RoundContainer = styled.div`
   display: flex;
   justify-content: space-evenly;
   margin: 40px;
 `;
-const StyledVersus = styled.span`
+const VSTag = styled.span`
   font-weight: 700;
   align-self: center;
 `;
-const StyledIcon = styled.img`
+const Icon = styled.img`
   width: 30px;
   padding-left: 5px;
 `;
 
 class DetailBattle extends Component {
   componentDidMount() {
-    const { setBattle, match } = this.props;
+    const { setBattle, match, battles } = this.props;
     const { id } = match.params;
-    setBattle(id);
+    if (battles === undefined) {
```

Comment:
Check if the battle from ID is present in the store, instead of "if there's any battle", for correctness.